bash_completion

```diff
@@ -2369,8 +2369,12 @@ fi
 unset compat_dir i _blacklist_glob
 
 # source user completion file
+#
+# Remark: We explicitly check that $user_completion is not '/dev/null' since
+#   /dev/null may be a regular file in broken systems and can contain arbitrary
+#   garbages of suppressed command outputs.
```

Comment:
I'm not sure what to think about this. On the other hand it seems quite benign, but on the other a broken system is a broken system, and if we start to add workarounds like this for specific cases, where does it end? What if the user completion points to some other file with random junk in it?

Reply:
Yeah, that is a valid concern. Broken systems are broken systems, so we don't have to care about them too much in order not to mess up the code. But at the same time, considering the facts that `/dev/null` as `BASH_COMPLETION_USER_FILE` may be frequently used to suppress user configurations, and also that it can possibly cause disasters in such a broken system though it is a rare case, I thought `/dev/null` might be treated as a special path.

In another way of thinking, `/dev/null` can be regarded as a special *keyword* to mean "*do nothing*". When users specify `/dev/null` to some variables, the users actually think about the behavior of this bit bucket but not the actual device linked to this particular location in the filesystem. In this sense, I don't think we should add more special rules to deal with other variations to represent the same location, such as `//dev/null`, `/tmp/../dev/null`, `/./././dev/null`, etc. (by using e.g. `readlink -f`) and also rules for other possible junk files `/dev/random`, `/dev/zero`, or whatever. The `/dev/null` is just special. I don't think there are so many other cases equally frequent and special as `/dev/null` for `BASH_COMPLETION_USER_FILE`. I suspect there are no such cases, so we don't have to worry about the endless workarounds.  We may add special treatment for this particular string just like `/dev/{tcp,udp}/$hostname/$port` in Bash or `/dev/{stdin,stdout,stderr,fd/$n}` in GNU awk.

I think that is also a valid perspective. I think it's not a matter that which one is correct or incorrect, so I'm not reluctant to drop this fix if you prefer not to have it in `bash_completion`! I'd like to hear your preference.

Reply:
I'm still kind of torn. It seems out of place in this PR now that we're no longer using `/dev/null` for `HISTFILE`. But nevermind, let's just keep it, it's a clean separate commit anyway.